main/src/com/google/refine/util/HttpClient.java

```diff
@@ -75,7 +80,113 @@ public HttpClient(int delay, int retryInterval) {
+                    .addRequestInterceptorFirst(new HttpRequestInterceptor() {
+
+                        private long nextRequestTime = System.currentTimeMillis();
+
+                        @Override
+                        public void process(
+                                final HttpRequest request,
+                                final EntityDetails entity,
+                                final HttpContext context) throws HttpException, IOException {
+
+                            long delay = nextRequestTime - System.currentTimeMillis();
+                            if (delay > 0) {
+                                try {
+                                    Thread.sleep(delay);
+                                } catch (InterruptedException e) {
+                                }
+                            }
+                            nextRequestTime = System.currentTimeMillis() + _delay;
+
+                        }
+                    });
+
+        } else if (System.getProperty("http.proxyHost") != null && System.getProperty("https.proxyPort") != null) {
+            proxyHost = System.getProperty("http.proxyHost");
+            proxyPort = Integer.parseInt(System.getProperty("https.proxyPort"));
+            proxy = new HttpHost("http", proxyHost, proxyPort);
+
+            httpClientBuilder = HttpClients.custom()
+                    .setProxy(proxy)
+                    .setUserAgent(RefineServlet.getUserAgent())
+                    .setDefaultRequestConfig(defaultRequestConfig)
+                    .setConnectionManager(connManager)
+                    // Default Apache HC retry is 1x @1 sec (or the value in Retry-Header)
+                    .setRetryStrategy(new ExponentialBackoffRetryStrategy(3, TimeValue.ofMilliseconds(_retryInterval)))
+//                    .setRedirectStrategy(new LaxRedirectStrategy()) // TODO: No longer needed since default doesn't exclude POST?
+//                   .setConnectionBackoffStrategy(ConnectionBackoffStrategy)
+                    .addRequestInterceptorFirst(new HttpRequestInterceptor() {
+
+                        private long nextRequestTime = System.currentTimeMillis();
+
+                        @Override
+                        public void process(
+                                final HttpRequest request,
+                                final EntityDetails entity,
+                                final HttpContext context) throws HttpException, IOException {
+
+                            long delay = nextRequestTime - System.currentTimeMillis();
+                            if (delay > 0) {
+                                try {
+                                    Thread.sleep(delay);
+                                } catch (InterruptedException e) {
+                                }
+                            }
+                            nextRequestTime = System.currentTimeMillis() + _delay;
+
+                        }
+                    });
```

Comment:
It looks like you have duplicated a lot of code here! I think there should be a way to rewrite this to minimize the code duplication.

Reply:
Am looking into it

Reply:
@elroykanye Could there be an issue with my PR. 3 checks have been in progress for hours